src/main/java/org/traccar/protocol/WialonProtocolDecoder.java

```diff
@@ -156,11 +154,52 @@ private Position decodePosition(Channel channel, SocketAddress remoteAddress, St
                     }
                 }
             }
+
+            if (position.hasAttribute("accuracy")) {
+                position.setAccuracy(position.removeDouble("accuracy"));
+            }
+
+            if (position.hasAttribute("bat")) {
+                position.set(Position.KEY_BATTERY_LEVEL, position.removeInteger("bat"));
+            }
+
+            if (position.hasAttribute("temp")) {
+                position.set(Position.KEY_DEVICE_TEMP, position.removeInteger("temp"));
+            }
+
+            Network network = new Network();
+            if (position.hasAttribute("mcc") || position.hasAttribute("mcc1")) {
```

Comment:
Why do we need this check?

Reply:
no particular reason apart of a few cpu cycles saved sometimes. Removed